examples/vision/learnable_resizer.py

```diff
@@ -0,0 +1,285 @@
+    for _ in range(num_res_blocks):
+        x = res_block(bottleneck)
+
+    # Projection.
+    x = layers.Conv2D(
+        filters=filters, kernel_size=3, strides=1, padding="same", use_bias=False
+    )(x)
+    x = layers.BatchNormalization()(x)
+
+    # Skip connection.
+    x = layers.Add()([bottleneck, x])
+
+    # Final resized image.
+    x = layers.Conv2D(filters=3, kernel_size=7, strides=1, padding="same")(x)
+    final_resize = layers.Add()([naive_resize, x])
+
+    return final_resize
+
+
+"""
+## Visualize the outputs of the learnable resizing module
+
+Here, we visualize how the resized images would look like after being passed through the
+random weights of the resizer. 
+"""
+
+sample_images, _ = next(iter(train_ds))
+
+plt.figure(figsize=(10, 10))
+for i, image in enumerate(sample_images[:9]):
+    ax = plt.subplot(3, 3, i + 1)
+    image = tf.image.convert_image_dtype(image, tf.float32)
+    resized_image = learnable_resizer(image[None, ...])
+    plt.imshow(resized_image.numpy().squeeze())
+    plt.axis("off")
+
+"""
+## Model building utility
+"""
+
+
+def get_model():
+    backbone = tf.keras.applications.DenseNet121(
+        weights=None,
+        include_top=True,
+        classes=2,
+        input_shape=((TARGET_SIZE[0], TARGET_SIZE[1], 3)),
+    )
+    backbone.trainable = True
+
+    inputs = layers.Input((INP_SIZE[0], INP_SIZE[1], 3))
+    x = layers.experimental.preprocessing.Rescaling(scale=1.0 / 255)(inputs)
+    x = learnable_resizer(x)
+    outputs = backbone(x)
+
+    return tf.keras.Model(inputs, outputs)
+
+
+"""
+The structure of the learnable image resizer module allows for flexible integrations with
+different vision models.
+"""
+
+"""
+## Compile and train our model with learnable resizer
+"""
+
+model = get_model()
+model.compile(
+    loss=keras.losses.CategoricalCrossentropy(label_smoothing=0.1),
+    optimizer="sgd",
+    metrics=["accuracy"],
+)
+model.fit(train_ds, validation_data=validation_ds, epochs=EPOCHS)
+
+"""
+## Visualize the outputs of the trained visualizer
+"""
+
+learned_resizer = tf.keras.Model(model.input, model.layers[-2].output)
+
+plt.figure(figsize=(10, 10))
+for i, image in enumerate(sample_images[:9]):
+    ax = plt.subplot(3, 3, i + 1)
+    image = tf.image.convert_image_dtype(image, tf.float32)
+    resized_image = learned_resizer(image[None, ...])
+    plt.imshow(resized_image.numpy().squeeze())
+    plt.axis("off")
+
+"""
+The plot shows that the visuals of the images have improved with training. The following
+table shows the benefits of using the resizing module in comparison to using the bilinear 
+interpolation:
+
+|           Model           	| Number of  parameters (Million) 	| Top-1 accuracy 	|
+|:-------------------------:	|:-------------------------------:	|:--------------:	|
+|   With the learnable resizer  	|             7.051717            	|      67.67%     	|
+| Without the learnable resizer 	|             7.039554            	|      60.19%      	|
```

Comment:
These results are pretty convincing, the increase in parameter count is small and the increase in accuracy is solid!

Reply:
Yes, exactly. I did run the experiments for like three times and each time I was able to get a better performance with the resizer. To be absolutely sure and reproducible, I trained the models with the same initial random weights (I have mentioned it in the example).